wdl/downsample.wdl

```diff
@@ -1,305 +1,284 @@
-
-            fastq-sample -n ${freads} --seed ${seed} -o ${downsample_file_2} ${fastq_file_2}
-
-        else
-            echo "Error: counts don't match up!"
-            echo $count1
-            echo $count2
-        fi
-    >>>
-
-    runtime {
-        docker: downsample_docker
-    }
-
-}
-
-task realign {
+  input {
+    File reference_fasta
+    String downsample_docker
+    File bam_or_cram_file
+
+    File ref_amb
+    File ref_ann
+    File ref_bwt
+    File ref_pac
+    File ref_sa
+    File ref_fai
+    File ref_dict
+
+    String sample_ID
+    File intervals_exons
+
+    #default start depth is 30x
+    Float start_depth = 30
+    Int? seed_override  
+
+    File? gatk4_jar_override
+    String gatk_docker
+
+    # Runtime configuration overrides
+    RuntimeAttr? runtime_attr_cram_to_bam
+    RuntimeAttr? runtime_attr_bam_to_fastq 
+    RuntimeAttr? runtime_attr_random_sample
+    RuntimeAttr? runtime_attr_realign
+    RuntimeAttr? runtime_attr_add_read_group
+    RuntimeAttr? runtime_attr_mark_duplicates
+    RuntimeAttr? runtime_attr_sort_index
+    RuntimeAttr? runtime_attr_count_coverage
+    RuntimeAttr? runtime_attr_collect_counts   
+
+    #Execution defaults and overrides
+    Boolean run_downsample_2x = true
+    Boolean run_downsample_4x = true
+    Boolean run_downsample_6x = true
+    Boolean run_downsample_8x = true
+    Boolean run_downsample_custom = false
+
+    Float? final_depth_custom
+
+  }
+
+  meta {
+    author: "Stephanie Hao"
+    email: "shao@broadinstitute.org"
+  }
     
-    input {
-        File downsample_file_1 
-        File downsample_file_2 
-        File reference_fasta
-        String downsample_docker
+  #################################################################################
+  ####        Calls part 1 to convert cram/bam file to paired fastq               #
+  #################################################################################
+
+  call ds1.downSampling_01 as downSampling_01 {
+    input :
+      bam_or_cram_file = bam_or_cram_file,
+      reference_fasta = reference_fasta,
+      downsample_docker = downsample_docker,
+      runtime_attr_cram_to_bam = runtime_attr_cram_to_bam,
+      runtime_attr_bam_to_fastq = runtime_attr_bam_to_fastq 
+  }
+
+  Float final_depth_2x = 2
+  Float final_depth_4x = 4
+  Float final_depth_6x = 6
+  Float final_depth_8x = 8
+
+  #################################################################################
+  ####        Downsamples samples to 2x, 4x, 6x, and 8x read depths               #
+  #################################################################################
+  if (run_downsample_2x) {
+    call ds2.downSampling_02 as downSampling_02_2x {
+      input :
+        fastq_file_1 = downSampling_01.fastq_1,
+        fastq_file_2 = downSampling_01.fastq_2,
+        downsample_docker = downsample_docker,
+        start_depth = start_depth,
+        final_depth = final_depth_8x,
```

Comment:
Should this be final_depth_2x? Also I think you said that you were testing this in pieces - it would be good to do a full run-through and check the count outputs to make sure everything ran correctly before you scale up, to make sure all the little things are squared away.

```suggestion
        final_depth = final_depth_2x,
```